Sources/SecureXPC/XPCError.swift

```diff
@@ -69,4 +69,38 @@ public enum XPCError: Error, Codable {
     case other(String)
     /// Unknown error occurred.
     case unknown
+    
+    /// Represents the provided error as an ``XPCError``.
+    ///
+    /// - Parameters:
+    ///   - error: The error to be represented as an ``XPCError``
+    ///   - expectingNonXPCError: Whether an error which is not an ``XPCError``, `DecodingError`, or `EncodingError` is expected. If it is,
+    ///   then its description will be returned as ``XPCError/other(_:)``, otherwise ``XPCError/unknown`` will be returned.
+    /// - Returns: An ``XPCError`` to represent the passed in `error`
+    internal static func asXPCError(error: Error, expectingOtherError: Bool) -> XPCError {
+        if let error = error as? XPCError {
+            return error
+        } else if let error = error as? DecodingError {
+            return .decodingError(String(describing: error))
+        } else if let error = error as? EncodingError {
+            return .encodingError(String(describing: error))
+        } else if expectingOtherError {
+            return .other(String(describing: error))
+        } else {
+            return .unknown
+        }
```

Comment:
Huh, why is this more idiomatic? The last two cases seem rather oddly constructed where a `_` is needed; particularly the penultimate one `case _ where expectingOtherError`.

Reply:
I'll commit this as-is, but happy to revise it later

Reply:
It's just a general trend in Swift code to prefer switch over if/else, because it makes it more clear that the predicates are all related (in that they're checking against the same operand).

A preference one way or the other isn't explicitly documented in any of the style guides I know, but some rationale is explained well here (in the context of C++, where their `switch` statements are waaaaaaay less powerful): https://youtu.be/raB_289NxBk?t=287

Granted, the last 2 cases here muddy the water a bit, though nonetheless I think they communicate intent very well